syntax = "proto3";

package audio;

option go_package = "internal/audioproto";

service AudioAnalyzer {
    rpc ProcessAudio (AudioRequest) returns (AudioResponse);
}

message AudioRequest {
    string audio_path = 1;
}

message AudioResponse {
    repeated float pitches = 1; // Основная частота (Гц)
    repeated float times = 2;   // Временные метки (сек)
    bytes chromagram = 3;       // Хромаграмма
    float tempo = 4;            // Темп
    float sr = 5;               // Частота дискретизации
    int32 hop_length = 6;       // Шаг
}